### PR TITLE
Adds applepay and googlepay support

### PR DESCRIPTION
**Pull request by alenad**

### CODE REVIEW

**Comment by alenad** [comment]
@daliwei @pierre

**Comment by pierre** [comment]
@alenad Could you add some tests?

**Comment by alenad** [comment]
> @alenad Could you add some tests?

@pierre tests added

File: src/main/java/org/killbill/billing/plugin/adyen/api/mapping/PaymentInfoMappingService.java [hunk 1]
```diff
@@ -134,6 +134,19 @@ private static void set3DSecureFields(final PaymentInfo paymentInfo, final Itera
...
+        final String mpiDataEci = PluginProperties.findPluginPropertyValue(PROPERTY_MPI_DATA_ECI, properties);
+        paymentInfo.setMpiDataEci(mpiDataEci);
+
+        final String selectedBrand = PluginProperties.findPluginPropertyValue(PROPERTY_SELECTED_BRAND, properties);
+
+        if (selectedBrand != null) {
+            paymentInfo.setMpiDataDirectoryResponse("Y");
+            paymentInfo.setMpiDataAuthenticationResponse("Y");
+            if (mpiDataEci == null || mpiDataEci.isEmpty()) {
+                paymentInfo.setMpiDataEci("7");
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/api/mapping/PaymentInfoMappingService.java]: Is it expected the brand overrides this value?

File: src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java [hunk 2]
```diff
@@ -107,6 +107,30 @@
...
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_USER_AGENT, "Java/1.8")
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_ACCEPT_HEADER, "application/json")
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_TERM_URL, "dummy://url")
+                                                                                                                  .build());
+    private final Iterable<PluginProperty> propertiesWithApplePay = PluginProperties.buildPluginProperties(ImmutableMap.<String, String>builder()
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_CC_LAST_NAME, "Applepaytester")
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_SELECTED_BRAND, "applepay")
```
Comment (pierre) [inline src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java]: Could you add a test with brand and without `eci` to verify the logic?

File: src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java [hunk 3]
```diff
@@ -107,6 +107,30 @@
...
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_MPI_DATA_CAVV, "AAAAAAAAt5fMJPDr320qAAALwwA=hq0BA9EAAAGXIJcAGAAAABKU0+s=")
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_MPI_DATA_ECI, "7")
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_CC_NUMBER, CC_3DS_NUMBER)
```
Comment (pierre) [inline src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java]: Why are you also passing a full PAN? Are you sure Adyen treats the request as an ApplePay one?

File: src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java [hunk 4]
```diff
@@ -107,6 +107,30 @@
...
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_CC_EXPIRATION_MONTH, String.valueOf(CC_EXPIRATION_MONTH))
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_CC_EXPIRATION_YEAR, String.valueOf(CC_EXPIRATION_YEAR))
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_USER_AGENT, "Java/1.8")
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_ACCEPT_HEADER, "application/json")
+                                                                                                                  .put(AdyenPaymentPluginApi.PROPERTY_TERM_URL, "dummy://url")
```
Comment (pierre) [inline src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java]: Probably not needed.

File: src/main/java/org/killbill/billing/plugin/adyen/api/mapping/PaymentInfoMappingService.java (hunk 1 above)
Comment (alenad) [reply src/main/java/org/killbill/billing/plugin/adyen/api/mapping/PaymentInfoMappingService.java]: Yes, Adyen asked to send these values.

File: src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java (hunk 3 above)
Comment (alenad) [reply src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java]: > Why are you also passing a full PAN? Are you sure Adyen treats the request as an ApplePay one?

Yes, treats as applepay.

**Pull request merged**